src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1221 @@
+  amResults.CuMinExecCycles[index]  |= currData[5];
+  amResults.CuMaxExecCycles[index]  |= currData[6];
+  amResults.CuStartCount[index]     |= currData[7];
+
+  if(hasDataflow) {
+    uint64_t dfTmp[2] = {0};
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_BUSY_CYCLES_OFFSET, &dfTmp[0], sizeof(uint32_t));
+    xclRead(handle, XCL_ADDR_SPACE_DEVICE_PERFMON, dbgIpInfo->m_base_address + XAM_MAX_PARALLEL_ITER_OFFSET, &dfTmp[1], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+
+    amResults.CuBusyCycles[index]      |= dfTmp[0] << 32;
+    amResults.CuMaxParallelIter[index] |= dfTmp[1] << 32;
+  } else {
+    amResults.CuBusyCycles[index]      = amResults.CuExecCycles[index];
+    amResults.CuMaxParallelIter[index] = 1;
+  }
+
+  (void)size;
+
+  // increment debugIpNum
+  ++debugIpNum[ACCEL_MONITOR];
+  amResults.NumSlots = (unsigned int)debugIpNum[ACCEL_MONITOR];
+}
+
+void 
+DebugIpStatusCollector::printAMResults(std::ostream& _output)
+{
+  if(0 == amResults.NumSlots) {
+    return;
+  }
+
+  _output << "\nAccelerator Monitor Counters (hex values are cycle count)\n";
+
+  auto col1 = std::max(cuNameMaxStrLen[ACCEL_MONITOR], strlen("Compute Unit")) + 4;
+
+  boost::format header("%-"+std::to_string(col1)+"s %-8s  %-8s  %-16s  %-16s  %-16s  %-16s  %-16s  %-16s  %-16s");
+  _output << header % "Compute Unit"  % "Ends" % "Starts" % "Max Parallel Itr" % "Execution" % "Memory Stall" % "Pipe Stall" % "Stream Stall" % "Min Exec" % "Max Exec"
+          << std::endl;
+
+  boost::format valueFormat("%-"+std::to_string(col1)+"s %-8llu  %-8llu  %-16llu  0x%-14x  0x%-14x  0x%-14x  0x%-14x  0x%-14x  0x%-14x");
+  for (size_t i = 0; i < amResults.NumSlots; ++i) {
+    _output << valueFormat
+                  % cuNames[ACCEL_MONITOR][i] 
+                  % amResults.CuExecCount[i] % amResults.CuStartCount[i] % amResults.CuMaxParallelIter[i] 
+                  % amResults.CuExecCycles[i] 
+                  % amResults.CuStallExtCycles[i] % amResults.CuStallIntCycles[i] % amResults.CuStallStrCycles[i]
+                  % amResults.CuMinExecCycles[i]  % amResults.CuMaxExecCycles[i]
+            << std::endl;
+  }
+}
+
+void 
+DebugIpStatusCollector::readASMCounter(debug_ip_data* dbgIpInfo)
+{
+  // index in results is debugIpNum
+  const uint64_t index = debugIpNum[AXI_STREAM_MONITOR];
+
+  // Get Debug Ip Name
+  std::string dbgIpName;
+  // Fill up string with 128 characters (padded with null characters)
+  dbgIpName.assign(dbgIpInfo->m_name, sizeof(dbgIpInfo->m_name));
+  // Strip away any extraneous null characters
+  dbgIpName.assign(dbgIpName.c_str());
+
+  // Get Master and Slave Port Name
+  std::string masterName;
+  std::string slaveName;
+  getStreamName(dbgIpInfo->m_type, dbgIpName, masterName, slaveName);
+  cuNames[AXI_STREAM_MONITOR].emplace_back(masterName);
```

Comment:
I think you mean to 'move' masterName and slaveName into the vectors, so do it this way
```
{
  std::string masterName;
  std::string slaveName;
  getStreamName(dbgIpInfo->m_type, dbgIpName, masterName, slaveName);
  cuNames[AXI_STREAM_MONITOR].emplace_back(std::move(masterName));
  portNames[AXI_STREAM_MONITOR].emplace_back(std::move(slaveName));
}
```
The scope ensures that masterName and slaveName are not used later, and the std::move creates a rvalue from the lvalue strings so that the string content is moved during emplace rather than copied.

Reply:
Done